### PR TITLE
Remove Runtime Reflection In Parser.

### PR DESCRIPTION
**Pull request by Josef-Vonasek**
### Pull Request Description

Removes runtime reflection from parser, enabling us to use scalajs.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: Syntax/definition/src/main/scala/org/enso/syntax/text/spec/DocParserDef.scala [hunk 1]
```diff
@@ -103,7 +101,7 @@ case class DocParserDef() extends Parser[Doc] {
     }
   }
 
-  ROOT || normalText || reify { text.onPushing(currentMatch) }
+  ROOT || normalText || text.onPushing(currentMatch)
```
Comment (BinarySoftware) [inline Syntax/definition/src/main/scala/org/enso/syntax/text/spec/DocParserDef.scala]: I really like the removal of reify, makes code a lot nicer to read

File: Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala [hunk 2]
```diff
@@ -54,6 +52,7 @@ class DocParser {
   //// HTML Rendering of Documentation /////////////////////////////////////////
   //////////////////////////////////////////////////////////////////////////////
 
+  // TODO remove this functionality from parser
```
Comment (BinarySoftware) [inline Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala]: Idk why you want to remove html generator, but if you have seen my PR it adds a lot to html renderer

Comment (Josef-Vonasek) [reply Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala]: According to @wdanilo parser isn't supposed to create any files. Furthermore we want to move to scalajs, which can't write or read files (at least no trough the java api).

Comment (Josef-Vonasek) [reply Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala]: I mean, I think it's fine for parser to return html string of documentation, but it doesn't make sense for it to also save it as file.

Comment (wdanilo) [reply Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala]: @BinarySoftware the idea is to remove files IO operations and leave HTML generator which results in string. Parser needs to compile to JS, with this code its impossible.

**Pull request merged**